src/interceptors/ClientRequest/NodeClientRequest.ts

```diff
@@ -163,6 +164,38 @@ export class NodeClientRequest extends ClientRequest {
     })
   }
 
+  emit(event: string, ...data: any[]) {
+    this.log('event:%s', event)
+
+    if (event === 'response') {
+      this.log('found "response" event, cloning the response...')
+
+      try {
+        /**
+         * Clone the response object when emitting the "response" event.
+         * This prevents the response body stream from locking
+         * and allows reading it twice:
+         * 1. Internal "response" event from the observer.
+         * 2. Any external response body listeners.
+         * @see https://github.com/mswjs/interceptors/issues/161
+         */
+        const response = data[0] as IncomingMessage
+        const firstClone = cloneIncomingMessage(response)
+        const secondClone = cloneIncomingMessage(response)
```

Comment:
Clone the original `IncomingMessage` (response) into two `PassThrough` streams so they can be read independently by:

- Internal `response` event emitted by the `observer` (contains `response.body` from `getIncomingMessageBody`);
- Any external `response` event listeners on `ClientRequest`.